spec/services/get_products_service_spec.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe GetProductsService do
+  subject(:service) { described_class.new(category_id) }
+
+  let(:all_category) { create(:category, name: 'all') }
+  let(:tea_category) { create(:category, parent: all_category, name: 'tea') }
+  let(:coffee_category) { create(:category, parent: all_category, name: 'coffee') }
+  let(:milk_tea_category) { create(:category, parent: tea_category, name: 'milk_tea') }
+  let(:hi_tea_category) { create(:category, parent: tea_category, name: 'hi_tea') }
+
+  let!(:americano) { create(:product, category: coffee_category) }
+  let!(:iced_coffee) { create(:product, category: coffee_category) }
+  let!(:hi_tea) { create(:product, category: hi_tea_category) }
+  let!(:milk_tea) { create(:product, category: milk_tea_category) }
+
+  describe '#validate' do
+    context 'when category_id does not exist in db' do
+      let(:category_id) { SecureRandom.uuid }
+
+      it 'returns error' do
+        service.call
+
+        expect(service.error?).to eq true
+        expect(service.first_error.message).to eq 'category is not found.'
+      end
+    end
+
+    describe '#call' do
+      context 'when a category is root' do
```

Comment:
root maybe unfamiliar to people does not know this approach, should use when category does not have parent